include/message_filters/subscriber.h

```diff
@@ -140,7 +154,43 @@ class Subscriber : public SubscriberBase, public SimpleFilter<M>
                [this](std::shared_ptr<M const> msg) {
                  this->cb(EventType(msg));
                }, qos);
-      node_ = node;
+
+      // The subscriber is switching to using a shared pointer, so relase the raw.
+      if (node_raw_ != nullptr) {
+        node_raw_ = nullptr;
+      }
+      node_shared_ = node;
```

Comment:
It would be nice, to find a way to deduplicate the code here, maybe having the `shared_ptr` version call the raw pointer version and then change which is stored in `node_raw_` versus `node_shared_`.

Reply:
Please take a look at the updated logic.